src/multiint.cpp

```diff
@@ -2161,6 +2174,60 @@ void WzMultiplayerOptionsTitleUI::closePositionChooser()
 	positionChooserUp = -1;
 }
 
+void WzMultiplayerOptionsTitleUI::openFactionChooser(uint32_t player)
+{
+	ASSERT_OR_RETURN(, player < MAX_PLAYERS, "Invalid player number");
+	initInlineChooser(player);
+
+	// add form.
+	auto psParentForm = (W_FORM *)widgGetFromID(psWScreen, MULTIOP_PLAYERS);
+	addInlineChooserBlueForm(psInlineChooserOverlayScreen, psParentForm, MULTIOP_FACCHOOSER_FORM, "",
+		7,
+		playerBoxHeight(player),
+		MULTIOP_ROW_WIDTH, MULTIOP_PLAYERHEIGHT);
+
+	// add the flags
+	int flagW = iV_GetImageWidth(FrontImages, IMAGE_FACTION_NORMAL) + 4;
+	int flagH = iV_GetImageHeight(FrontImages, IMAGE_PLAYERN);
+	int space = MULTIOP_ROW_WIDTH - 0 - flagW * NUM_FACTIONS;
+	int spaceDiv = NUM_FACTIONS;
+	space = std::min(space, 5 * spaceDiv);
+
+	auto psWeakTitleUI = std::weak_ptr<WzMultiplayerOptionsTitleUI>(std::dynamic_pointer_cast<WzMultiplayerOptionsTitleUI>(shared_from_this()));
+
+	for (unsigned int i = 0; i < NUM_FACTIONS; i++)
+	{
+		auto onClickHandler = [player, psWeakTitleUI](W_BUTTON &button) {
+			UDWORD id = button.id;
+			auto pStrongPtr = psWeakTitleUI.lock();
+			ASSERT_OR_RETURN(, pStrongPtr.operator bool(), "WzMultiplayerOptionsTitleUI no longer exists");
+
+			STATIC_ASSERT(MULTIOP_FACCHOOSER + NUM_FACTIONS - 1 <= MULTIOP_FACCHOOSER_END);
+			if (id >= MULTIOP_FACCHOOSER && id <= MULTIOP_FACCHOOSER + NUM_FACTIONS -1)
+			{
+				// TODO: @cpdef clean this
+				resetReadyStatus(false, true);
+				uint8_t idx = id - MULTIOP_FACCHOOSER;
+				SendFactionRequest(player, idx);
+//				NetPlay.players[player].faction = static_cast<FactionID>(idx);
+//				NETBroadcastPlayerInfo(player);
```

Comment:
Presumably we can remove these commented-out lines now? (As well as the TODO above?)